test/runtests.jl

```diff
@@ -1,30 +1,59 @@
 using
+    Oceananigans,
+    Oceananigans.Operators,
+    Oceananigans.TurbulenceClosures,
     Test,
+    Random,
+    JLD2,
+    Printf,
     Statistics,
-    OffsetArrays
+    OffsetArrays,
+    FFTW
 
-import FFTW
+@hascuda using CuArrays
 
-using
-    Oceananigans,
-    Oceananigans.Operators,
-    Oceananigans.TurbulenceClosures
+using Statistics: mean
+using LinearAlgebra: norm
+using GPUifyLoops: @launch, @loop
 
 using Oceananigans: PoissonSolver, PPN, PNN, solve_poisson_3d!,
-                    parentdata, buoyancy_perturbation, fill_halo_regions!, run_diagnostic
+                    velocity_div!, compute_w_from_continuity!,
+                    launch_config, datatuples, device,
+                    parentdata, fill_halo_regions!, run_diagnostic,
+                    TracerFields, buoyancy_frequency_squared, thermal_expansion, haline_contraction, ρ′,
+                    RoquetIdealizedNonlinearEquationOfState
+
+import Oceananigans: datatuple
+
+using Oceananigans.TurbulenceClosures: ∂x_caa, ∂x_faa, ∂x²_caa, ∂x²_faa,
+                                       ∂y_aca, ∂y_afa, ∂y²_aca, ∂y²_afa,
+                                       ∂z_aac, ∂z_aaf, ∂z²_aac, ∂z²_aaf,
+                                       ▶x_caa, ▶x_faa, ▶y_aca, ▶y_afa,
+                                       ▶z_aac, ▶z_aaf
+
+float_types = (Float32, Float64)
 
 archs = (CPU(),)
 @hascuda archs = (CPU(), GPU())
-@hascuda using CuArrays
 
-float_types = (Float32, Float64)
+closures = (
+            :ConstantIsotropicDiffusivity,
+            :ConstantAnisotropicDiffusivity,
+            :SmagorinskyLilly,
+            :BlasiusSmagorinsky,
+            :RozemaAnisotropicMinimumDissipation,
+            :VerstappenAnisotropicMinimumDissipation
+           )
+
+EquationsOfState = (LinearEquationOfState, RoquetIdealizedNonlinearEquationOfState)
 
 @testset "Oceananigans" begin
     include("test_grids.jl")
     include("test_fields.jl")
     include("test_halo_regions.jl")
     include("test_operators.jl")
     include("test_poisson_solvers.jl")
+    include("test_coriolis_buoyancy.jl")
```

Comment:
Doesn't have to happen in this PR but would be good to split into two files `test_coriolis.jl` and `test_buoyancy.jl`